hummingbot/strategy/pure_market_making/pure_market_making_v2.pyx

```diff
@@ -295,10 +318,40 @@ cdef class PureMarketMakingStrategyV2(StrategyBase):
                                                                                    market_info,
                                                                                    market_info_to_active_orders.get(market_info, []),
                                                                                    orders_proposal)
+                filtered_proposal = self.c_filter_orders_proposal_for_takers(market_info, filtered_proposal)
                 self.c_execute_orders_proposal(market_info, filtered_proposal)
         finally:
             self._last_timestamp = timestamp
 
+    # To filter out any orders that are going to be taker orders, i.e. buy order price higher than first ask
+    # and sell order price lower than first bid on the order book.
+    cdef object c_filter_orders_proposal_for_takers(self, object market_info, object orders_proposal):
+        cdef :
+            list buy_prices = []
+            list buy_sizes = []
+            list sell_prices = []
+            list sell_sizes = []
+        if orders_proposal.buy_order_sizes[0] > 0:
+            for i in range(0, len(orders_proposal.buy_order_prices)):
```

Comment:
This is not necessary here. We should use Python idiomatic loops whenever possible.

e.g.
```
for buy_price in orders_proposal.buy_order_prices:
  ...
```

Reply:
I understand, but I need the index (i) for later accessing the buy_sizes array.
` buy_sizes.append(orders_proposal.buy_order_sizes[i]) `